src/orders/views/OrderReturn/utils.tsx

```diff
@@ -63,10 +93,39 @@ class ReturnFormDataParser {
 
       return [
         ...result,
-        ({ [idKey]: id, quantity, replace: shouldReplace } as unknown) as T,
+        { [idKey]: id, quantity, replace: shouldReplace } as unknown as T,
       ];
     }, []);
   };
+
+  private getShouldRefund = (
+    orderLines: OrderReturnLineInput[],
+    fulfillmentLines: OrderReturnFulfillmentLineInput[],
+  ) => {
+    if (
+      !this.order.totalCaptured?.amount ||
+      this.formData.amountCalculationMode ===
+        OrderRefundAmountCalculationMode.NONE
+    ) {
+      return false;
+    }
+
+    if (!!this.getAmountToRefund()) {
+      return true;
+    }
+
+    return (
+      orderLines.some(ReturnFormDataParser.isLineRefundable) ||
+      fulfillmentLines.some(ReturnFormDataParser.isLineRefundable)
+    );
+  };
+
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  private static isLineRefundable = function <
+    T extends OrderReturnLineInput | OrderReturnFulfillmentLineInput,
+  >({ replace }: T) {
+    return !replace;
+  };
```

Comment:
**suggestion:** this is pretty complex logic which should probably be unit tested

Reply:
I won't be adding unit tests for that, since it's a restored file from `main` branch that I've previously deleted 😅